app/upload/views.py

```diff
@@ -72,3 +82,22 @@ def upload_document(service_id):
             'mimetype': mimetype,
         }
     ), 201
+
+
+def _upload_document_with_hashed_recipient_email(verification_email, service_id, file_data, mimetype):
+    hasher = PasswordHasher(
+        memory_cost=15360,
+        time_cost=2,
+        parallelism=1,
```

Comment:
Nice, thanks for grabbing these from the https://cheatsheetseries.owasp.org/cheatsheets/Password_Storage_Cheat_Sheet.html#password-hashing-algorithms. Can we add a comment including the URL so it's really obvious where we've got these params from?

And we should also probably add a note saying that if we want to change the params, we might want to consider how (if) we migrate existing hashes - which is described here: https://cheatsheetseries.owasp.org/cheatsheets/Password_Storage_Cheat_Sheet.html#upgrading-legacy-hashes

Reply:
I've added the first link (pending git push). Since the second link is to the same page, I will just mention the idea behind it without pasting it in.